packages/binding-wotfirestore/README.md

```diff
@@ -0,0 +1,221 @@
+            console.log(s)
+          })
+        })
+      })
+    } catch (err) {
+      console.error('Script error:', err)
+    }
+  })
+  .catch((err) => {
+    console.error('Fetch error:', err)
+  })
+```
+
+### Server Example
+
+The server example produces a thing that allows for setting a property `count`. The thing is reachable via Firestore.
+
+`node example-server.js`
+
+```js
+// example-server.js
+const Servient = require('@node-wot/core').Servient
+const WoTFirestoreServer =
+  require('@hidetak/binding-wotfirestore').WoTFirestoreServer
+const WoTFirestoreCodec =
+  require('@hidetak/binding-wotfirestore').WoTFirestoreCodec
+
+const firestoreConfig = require('./firestore-config.json')
+
+// create server
+const server = new WoTFirestoreServer(firestoreConfig)
+
+// create Servient add Firebase binding
+let servient = new Servient()
+servient.addServer(server)
+
+const codec = new WoTFirestoreCodec()
+servient.addMediaType(codec)
+
+servient.start().then((WoT) => {
+  WoT.produce({
+    '@context': 'https://www.w3.org/2019/wot/td/v1',
+    title: 'MyCounter',
+    properties: {
+      count: {
+        type: 'integer',
+        observable: true,
+        readOnly: false
+      }
+    }
+  }).then((thing) => {
+    console.log('Produced ' + thing.getThingDescription().title)
+    thing.writeProperty('count', 0)
+
+    thing.expose().then(() => {
+      console.info(thing.getThingDescription().title + ' ready')
+      console.info('TD : ' + JSON.stringify(thing.getThingDescription()))
+      thing.readProperty('count').then((c) => {
+        console.log('count is ' + c)
+      })
+    })
+  })
+})
+```
+
+## How to use Firestore
+
+This Binding realizes the interaction between Client and Server for Property, Action, and Event by storing data in the following reference.
+
+- things/\<hostName\>%2F\<title of Thing Description\>  
+  Stores the Thing Description that the Server program exposes.
+- things/\<hostName\>%2F\<title of Thing Description\>%2Fproperties%2F\<name of property\>  
+  Stores the value of the Property when it is changed by the Server.
+- things/\<hostName\>%2F\<title of Thing Description\>%2FpropertyReceives%2F\<name of property\>  
+  Stores the value of the Property when it is changed by the Client.
+- things/\<hostName\>%2F\<title of Thing Description\>%2Factions%2F\<name of action\>  
+  Stores the action calling information when the Client invokes an Action.
+- things/\<hostName\>%2F\<title of Thing Description\>%2FactionResults%2F\<name of action\>  
+  Stores the result of an Action executed by the Server.
+- things/\<hostName\>%2F\<title of Thing Description\>%events%2F\<name of event\>  
+  Stores the emitted events.
+
+Data will always be overwritten and no history will be retained.
+
+## Support functions for Client and Server implementations
+
+The functions that can be implemented for Client and Server differ as follows.
+
+| functions          | Client | Server |
```

Comment:
Nitpicky comments:
- function should be called WoT operation
- read property should be readproperty, same for all others
- Having - for some operations gives the impression that they are not implemented however a client cannot emit an event so I think it is better to say n/a for that.

Other comments:
- Are meta operations like `readallproperties` implemented ? I am guessing not but it would be good to document it here.

Reply:
@egekorkan 
Thank you for giving comment !

To begin with, I may not yet understand the changes in the API that ExposedThing has.
Does this mean that the only methods available when implementing a Thing will be the following methods?
https://www.w3.org/TR/wot-scripting-api/#the-exposedthing-interface

If so, I don't think this table I've written is appropriate.

I think that there is a method that has not been implemented among the methods defined here, but when is it going to be implemented ?

Reply:
@egekorkan 
I deleted the table once because it might have been misleading.

Reply:
Sorry about checking everything this late. I think that this table is needed to give a quick overview of what different types of operations are supported by this binding. A Thing, regardless of its protocol, might need to offer events and it should be clear that I can use this binding to achieve that behavior I wanted. You can find the list of all the current operations at https://w3c.github.io/wot-architecture/#sec-hypermedia-forms

Reply:
@egekorkan 
Thank you for your comment.
I understood that you advised that I should describe the support methods about Consumed Thing.
In the table that I made, the method that Consumed Thing and Exposed Thing have was described in the same table, but because there is a difference in the method that Consumed Thing and Exposed Thing can use as shown below, I thought that it was not an appropriate table.
https://www.w3.org/TR/wot-scripting-api/#the-exposedthing-interface

I'm going to describe the methods that Consumed Thing supports.

Reply:
@hidetak thank you for putting these. That is exactly what I had meant! :)

Reply:
@egekorkan 
I added explanation supported method for Client(Consumed Thing).
https://github.com/eclipse/thingweb.node-wot/pull/417/files#diff-a4441d0ef49c86a2f95d61efe5a084b05169a1339cd419f0df1295733e436890R202
The explanation about the Server has been removed. This is because I don't understand clearly what methods should be supported by Exposed Thing.

Reply:
So I would say for an exposed thing, it is about whether your implementation can produce an Exposed Thing that has, let's say, that has writeable properties, i.e. a form that has op value of `writeproperty`.

Reply:
@egekorkan 
Thank you for giving comment.
If that is the case, I believe that I can express it in this table.